src/main/java/org/reaktivity/nukleus/http_cache/internal/stream/HttpCacheProxyNonCacheableResponse.java

```diff
@@ -0,0 +1,275 @@
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package org.reaktivity.nukleus.http_cache.internal.stream;
+
+import org.agrona.DirectBuffer;
+import org.reaktivity.nukleus.function.MessageConsumer;
+import org.reaktivity.nukleus.http_cache.internal.proxy.request.Request;
+import org.reaktivity.nukleus.http_cache.internal.types.HttpHeaderFW;
+import org.reaktivity.nukleus.http_cache.internal.types.ListFW;
+import org.reaktivity.nukleus.http_cache.internal.types.OctetsFW;
+import org.reaktivity.nukleus.http_cache.internal.types.stream.AbortFW;
+import org.reaktivity.nukleus.http_cache.internal.types.stream.BeginFW;
+import org.reaktivity.nukleus.http_cache.internal.types.stream.DataFW;
+import org.reaktivity.nukleus.http_cache.internal.types.stream.EndFW;
+import org.reaktivity.nukleus.http_cache.internal.types.stream.HttpBeginExFW;
+import org.reaktivity.nukleus.http_cache.internal.types.stream.ResetFW;
+import org.reaktivity.nukleus.http_cache.internal.types.stream.WindowFW;
+
+import static java.lang.System.currentTimeMillis;
+import static org.reaktivity.nukleus.http_cache.internal.HttpCacheConfiguration.DEBUG;
+import static org.reaktivity.nukleus.http_cache.internal.stream.util.HttpHeadersUtil.getHeader;
+
+final class HttpCacheProxyNonCacheableResponse extends HttpCacheProxyResponse
+{
+    private final HttpCacheProxyFactory streamFactory;
+
+    private MessageConsumer streamState;
+
+    private final MessageConsumer connectReplyThrottle;
+    private final long connectRouteId;
+    private final long connectReplyStreamId;
+
+    private Request streamCorrelation;
+    private long acceptInitialId;
+    private int acceptReplyBudget;
+
+    private long groupId;
+    private int padding;
+
+    private long traceId;
+
+    HttpCacheProxyNonCacheableResponse(
+        HttpCacheProxyFactory httpCacheProxyFactory,
+        MessageConsumer connectReplyThrottle,
+        long connectRouteId,
+        long connectReplyId,
+        long acceptInitialId)
+    {
+        this.streamFactory = httpCacheProxyFactory;
+        this.connectReplyThrottle = connectReplyThrottle;
+        this.connectRouteId = connectRouteId;
+        this.connectReplyStreamId = connectReplyId;
+        this.acceptInitialId = acceptInitialId;
+        this.streamState = this::beforeBegin;
+    }
+
+    void handleStream(
+        int msgTypeId,
+        DirectBuffer buffer,
+        int index,
+        int length)
+    {
+        streamState.accept(msgTypeId, buffer, index, length);
+    }
+
+    @Override
+    public String toString()
+    {
+        return String.format("%s[connectRouteId=%016x, connectReplyStreamId=%d, acceptReplyBudget=%016x, " +
+                " padding=%d]", getClass().getSimpleName(),
+            connectRouteId, connectReplyStreamId, acceptReplyBudget, padding);
+    }
+
+    private void beforeBegin(
```

Comment:
Remove this method and move `BeginFW.TYPE_ID` to unified `switch` in `onResponseMessage`.